### PR TITLE
Restore changes

### PR DESCRIPTION
**Pull request by richardcase**
Work in progress changes for the restore. 

Started the change to generate a single yaml file with all the yaml for a specific namespace.

### CODE REVIEW

File: utilslib/dr.py [hunk 1]
```diff
@@ -91,10 +93,13 @@ def parse_key(key):
         tuble containing the fields in the key based on '/' seperator.
         """
         fields = key.split('/')
-        if len(fields) != 4:
+        if len(fields) == 4:
+            return fields[0], fields[1], fields[2], fields[3]
```
Comment (nab-gha) [inline utilslib/dr.py:97]: I think we are expecting 5 fields? 
cluster
namespace
kind
version
name

File: utilslib/dr.py [hunk 2]
```diff
@@ -91,10 +93,13 @@ def parse_key(key):
...
+        elif len(fields) == 5:
+            return fields[0], fields[1], fields[2], fields[4]
+        else:
             raise Exception(
                 "key should comprise /cluster/namespace/kind/name")
```
Comment (nab-gha) [inline utilslib/dr.py:102]: no leading / and version (with / changed to _) after kind

File: utilslib/dr.py (hunk 1 above)
Comment (richardcase) [reply utilslib/dr.py:97]: Yes we are but the s3 bucket has a mixture of both at present and so i'm just allowing both until we clear up s3.

File: utilslib/dr.py (hunk 2 above)
Comment (richardcase) [reply utilslib/dr.py:102]: True, i've not cleard this up yet but will do.

**Review by nab-gha — APPROVED** [review]

**Pull request merged**